ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tensor.py

```diff
@@ -70,10 +73,266 @@ def _check_query(query):
     )
 
 
+def _helper_init_tensorarray(backend_fw, l_kwargs, fn=None):
+    id_write, kwargs = l_kwargs
+    with BackendHandler.update_backend(backend_fw) as ivy_backend:
+        local_importer = ivy_backend.utils.dynamic_import
+        function_module = local_importer.import_module(
+            "ivy.functional.frontends.tensorflow"
+        )
+        ta_frontend = function_module.tensor.TensorArray(**kwargs)
+        ta = tf.TensorArray(**kwargs)
+        if fn == "unstack":
+            ta = ta.unstack(tf.constant(id_write))
+            ta_frontend = ta_frontend.unstack(function_module.constant(id_write))
+        elif fn == "split":
+            ta = ta.split(**id_write)
+            ta_frontend = ta_frontend.split(**id_write)
+        elif fn == "scatter":
+            indices, value = [*zip(*id_write)]
+            ta = ta.scatter(indices, tf.cast(tf.stack(value), dtype=ta.dtype))
+            value = function_module.stack(list(map(function_module.constant, value)))
```

Comment:
Why do we generate `value` differently for our frontend? Since we are doing `tf.cast(tf.stack(value), dtype=ta.dtype)`, I would then expect to do `function_module.cast(function_module.stack(value), dtype=ta.dtype)`.

Reply:
Yes so the reason for doing that was, 
```
import tensorflow as tf
import ivy
import ivy.functional.frontends.tensorflow as tf_frontend
import numpy as np

arr = [np.array([1,2])]*2

tf.stack(arr)  #works

ivy.set_backend("torch")
tf_frontend.stack(arr) #raises error
tf_frontend.stack(list(map(tf_frontend.constant, arr))) # this gives same result
```

Reply:
From a functional prespective they both are achieving the same thing, though I guess we can still add `function_module.cast`